evennia/web/api/tests.py

```diff
@@ -0,0 +1,133 @@
+"""Tests for the REST API"""
+from evennia.utils.test_resources import EvenniaTest
+from evennia.web.api import serializers
+from rest_framework.test import APIClient
+from django.urls import reverse
+from django.test import override_settings
+from collections import namedtuple
+from django.conf.urls import url, include
+from django.core.exceptions import ObjectDoesNotExist
+
+urlpatterns = [
+    url(r"^", include("evennia.web.website.urls")),
+    url(r"^api/", include("evennia.web.api.urls", namespace="api")),
+]
+
+
+@override_settings(
+    REST_API_ENABLED=True, ROOT_URLCONF=__name__, AUTH_USERNAME_VALIDATORS=[]
+)
+class TestEvenniaRESTApi(EvenniaTest):
+    client_class = APIClient
+
+    def setUp(self):
+        super().setUp()
+        self.account.is_superuser = True
+        self.account.save()
+        self.client.force_login(self.account)
+
+    def tearDown(self):
+        try:
+            super().tearDown()
+        except ObjectDoesNotExist:
+            pass
+
+    def get_view_details(self, action):
+        """Helper function for generating list of named tuples"""
+        View = namedtuple("View", ["view_name", "obj", "list", "serializer", "create_data"])
+        views = [
+            View("object-%s" % action, self.obj1, [self.obj1, self.char1, self.exit, self.room1, self.room2, self.obj2,
+                                                   self.char2], serializers.ObjectDBSerializer,
+                 {"db_key": "object-create-test-name"}),
+            View("character-%s" % action, self.char1, [self.char1, self.char2], serializers.ObjectDBSerializer,
+                 {"db_key": "character-create-test-name"}),
+            View("exit-%s" % action, self.exit, [self.exit], serializers.ObjectDBSerializer,
+                 {"db_key": "exit-create-test-name"}),
+            View("room-%s" % action, self.room1, [self.room1, self.room2], serializers.ObjectDBSerializer,
+                 {"db_key": "room-create-test-name"}),
+            View("script-%s" % action, self.script, [self.script], serializers.ScriptDBSerializer,
+                 {"db_key": "script-create-test-name"}),
+            View("account-%s" % action, self.account2, [self.account, self.account2], serializers.AccountDBSerializer,
+                 {"username": "account-create-test-name"}),
+        ]
+        return views
+
+    def test_retrieve(self):
+        views = self.get_view_details("detail")
+        for view in views:
+            with self.subTest(msg="Testing {} retrieve".format(view.view_name)):
+                view_url = reverse(
+                    "api:{}".format(view.view_name), kwargs={"pk": view.obj.pk}
+                )
+                response = self.client.get(view_url)
+                self.assertEqual(response.status_code, 200)
```

Comment:
It would be good if this test also included an assertion about just what is returned as well. That it has a 200 status only means itreturned _something_ without dying in the process. This is true for some other tests below as well, might not point each one out explicitly.